ui/src/app/components/app-router.tsx

```diff
@@ -38,21 +38,36 @@ const NavRedirect = ({path}) => {
   return null;
 };
 
-export const AppRoute = ({path, data = {}, component: Component}): React.ReactElement => {
+export const AppRoute = ({path, data = {}, guards = [], component: Component}): React.ReactElement => {
   const routeParams = useParams();
   const routeHistory = useHistory();
 
-  return <Route exact={true} path={path} >
-    <Component urlParams={routeParams} routeHistory={routeHistory} routeConfig={data}/>
+  return <Route exact={true} path={path} render={
+    () => {
+      const { redirectPath = null } = fp.find(({allowed}) => !allowed(), guards) || {};
+      return redirectPath
+        ? <NavRedirect path={redirectPath}/>
+        : <Component urlParams={routeParams} routeHistory={routeHistory} routeConfig={data}/>;
+    }}>
   </Route>;
 };
 
 export const ProtectedRoutes = (
   {guards, children}: {guards: Guard[], children: any }): React.ReactElement => {
-  const { redirectPath = null } = fp.find(({allowed}) => !allowed(), guards) || {};
 
-  return redirectPath ? <NavRedirect path={redirectPath}/>
-  : <Fragment>{children}</Fragment>;
+  // Pass the guards to the individual routes. Be sure not to overwrite any existing guards
+  const guardedChildren = fp.flow(
+    fp.flatten,
+    fp.toPairs,
+    fp.map(
+      ([key, element]: [string, React.ReactElement]) => {
+        const {guards: elementGuards = []} = element.props;
+        return React.cloneElement(element, {key, guards: [...elementGuards, ...guards]});
+      }
+    )
+  )([children]);
```

Comment:
I gather children is a map of keys -> react elements. I'm a bit confused though, since the previous approach rendered this directly, and your new approach appears to instead render an array of elements. Does React just support either?

Reply:
`children` can be a react element or an array of react elements. React does support arrays of elements.

Reply:
I updated the type for `children` to help clarify this in the code and I added a comment.
And to expand on my previous comment - children can be many different types in react, but for the purposes of ProtectedRoutes - it should be a react element or array of elements. I could make it more specifically an `AppRoute` / array of routes. But I would like to leave that as an exercise for a future PR to work on my Typescript.